swagger.yaml

```diff
@@ -8,6 +8,59 @@ security:
   - bearerAuth: []
 
 paths:
+  "/users/{userId}/vacation":
+    put:
+      operationId: updateUserVacation
+      summary: Update user vacation days
+      description: Update total and remaining vacation days for a specific user by year.
+      tags:
+        - Update vacation
+      parameters:
+        - name: userId
+          in: path
+          required: true
+          description: The Keycloak ID of the user to update vacation for
+          schema:
+            type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                vacationDays:
+                  type: object
+                  additionalProperties:
+                    type: object
+                    properties:
+                      total:
+                        type: integer
+                      remaining:
+                        type: integer
+                  description: Vacation days by year
+              required:
+                - vacationDays
+      responses:
+  "200":
+    description: Successfully updated the vacation days
+    content:
+      application/json;charset=utf-8:
+        schema:
+          type: object
+          properties:
+            id:
+              type: string
+              description: The Keycloak ID of the updated user
+            vacationDays:
+              type: object
+              additionalProperties:
```

Comment:
This seems  a bit strange to me that we have an object with id and vacationdays, which is an object containing additional properties. Perhaps we need more details here? Why is it called additional properites, is it meant't to be named in such a way to indicate the year it represents? If so this needs to be made clear.

Reply:
I would suggest making a schema for this (see other schemas in the spec) rather than defining it "inline" here.